src/Shared/ThrottlingDomainEventBehavior.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading.RateLimiting;
+using Microsoft.Extensions.Options;
+
+namespace Workleap.DomainEventPropagation;
+
+internal class ThrottlingDomainEventBehavior(IOptions<EventPropagationThrottlingOptions> options) : ISubscriptionDomainEventBehavior, IDisposable
```

Comment:
nit: Old IDP habit: can we seal any class that isn't expected to be inherited from? Also applies to other new types, like the new options, options validator, etc.